feat(token mgmt): update main wallet view layout with community assets

caybro opened the pull request:
- update the CollectiblesView.qml view and delegates according to the latest design
- add CollectiblesView to Storybook
- add new section for Community Collectibles
- (re)use the community badge with tooltip and link action to take the user to the respective community
- add Community Collectibles info icon + popup
- create context menu for token delegates with actions (Send/Receive/Manage tokens/Hide)
- add confirmation popups when hiding a single or all community tokens
- emit a toast bubble after hiding the token(s)
- fix eliding the community name inside ManageTokensCommunityTag
- some smaller fixes/cleanups

Fixes #12519

### What does the PR do

Updates the Collectibles section UI of the main wallet page, adds support for displaying community collectibles and the ability to hide token(s)

### Affected areas

Wallet RightTabView, CollectiblesView, CollectibleView (delegate)

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

![image](https://github.com/status-im/status-desktop/assets/5377645/43edec3d-0fc1-4b38-97d1-31092a59e03f)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 7b3021f3 | [#1](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-12917/1/) | 2023-11-29 16:01:14 | ~5 min | `tests/nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/prs/job/linux/job/x86_64/job/tests-nim/job/PR-12917/1//consoleText) |

noeliaSD reviewed (commented):
Nice job!

I've added some comments / suggestions and only tested in `storybook` bc I don't have currently a complete setup to test inside the app. I was thinking maybe this pr is a good candidate for the `QA team` to test since probably they have a proper setup, wdyt?

caybro commented on ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:
```diff
@@ -1,105 +1,374 @@
+                ExpressionFilter {
+                    expression: {
+                        d.controller.settingsDirty
+                        return d.symbolIsVisible(model.symbol) && !model.communityId
```
I don't get the question :)

noeliaSD replied:
Discussed offline! :)

noeliaSD replied:
Agreed on changing the method name to `tokenIsVisible` or `collectibleIsVisible`.

caybro replied:
> Agreed on changing the method name to `tokenIsVisible` or `collectibleIsVisible`. If there is the same method for `assets`, rename it also there!

Yup, already done locally for both

caybro commented on ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:
```diff
@@ -1,105 +1,374 @@
+        readonly property var communityCollectiblesModel: SortFilterProxyModel {
+            sourceModel: d.renamedModel
+            filters: [
+                ExpressionFilter {
```
I was thinking about this but couldn't find any alternative :/

caybro replied:
That said, I will at least fold it into one SFPM model in the followup PR (combined with the other sort criteria)

noeliaSD replied:
Thanks to recheck!

caybro commented on ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:
```diff
@@ -1,105 +1,374 @@
+        spacing: 0
+
+        ShapeRectangle {
+            visible: !d.hasCollectibles
```
This is with no collectibles at all

caybro replied:
Ah, I stand corrected, it's the regular collectibles; I can extend that condition to cover both, wdyt?

noeliaSD replied:
Yeah, definitly I think, at least, we need to ask @benjthayer how it should look like in all these options!

caybro replied:
Fixed, now it checks both

caybro commented on ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:
```diff
@@ -1,105 +1,374 @@
+            text: qsTr("Collectibles will appear here")
+        }
+
+        CustomGridView {
```
Not in the current designs (that's the next iteration ;)

caybro commented on storybook/pages/CollectiblesViewPage.qml:
It already does the event logging (the Events tab on right):
![image](https://github.com/status-im/status-desktop/assets/5377645/47c44a01-18a1-432f-aa65-606a7c068ebe)

If you go to ManageCollectiblesPanel SB page, you can clear the settings from there:
![image](https://github.com/status-im/status-desktop/assets/5377645/5520f9d8-15cf-48a6-9b60-038e1092ccc0)

And the warnings:
Yes, the mock models simply don't have the methods that the real ones do, nothing I can do about it here :/

noeliaSD replied:
- Oh got it ! I didn't see the events log before! Prefect then!
- For the action of clearing settings.. that's good if we have this option in the manage collectibles panel BUT I'm not sure about that interconnections.. From mpov the pages should be self-contained.

caybro commented on storybook/pages/CollectiblesViewPage.qml:
```diff
@@ -0,0 +1,69 @@
+        logsView.logText: logs.logText
+
+        ColumnLayout {
+            Switch {
```
Ah hmm... good suggestion; can you pls file a separate issue for that? :) It's certianly something that could be done in SB only using SFPM I think

noeliaSD replied:
https://github.com/status-im/status-desktop/issues/12948

caybro commented on ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:
```diff
@@ -1,105 +1,374 @@
+            sorters: [
+                RoleSorter {
+                    roleName: "name"
+                    enabled: !d.isCustomView
```
Yeah, the enum is coming with the next PR :) (on my disc currently)

caybro replied:
See the example above where `cmbTokenOrder` is the sort combo box

alexjba reviewed (commented):
Nice work! LGTM in general!

Few findings:

- The button text needs to be elided

<img width="807" alt="Screenshot 2023-12-05 at 15 42 56" src="https://github.com/status-im/status-desktop/assets/47811206/a2be5b47-8f24-45f9-873c-f0ce1da62d5a">

- I've added a few watch accounts and I see some popular NTFs use webp encoding. It seems we can't play webp images from the network.
I've been briefly looked into this an I think this is the issue https://bugreports.qt.io/browse/QTBUG-75044. It was fixed in qt6, so it's not helpful in our case.
If we really need to support webp formats, I think we need an alternative. It seems to be working fine if we're downloading the image before loading it in the StatusAnimatedImage. @caybro WDYT of this?

```
Warning: qrc:/StatusQ/Components/StatusAnimatedImage.qml:24:1: QML StatusAnimatedImage: Error Reading Animated Image File https://i.seadn.io/gae/TX0qOy8g8ibx1p2gQzWIM3Kvsy4a447ti2ZtMlcJmRFM8PU_4YYw2WMpKvO_bnqJiDj2Y7njP6d0tf2YG_K-yYDN2gdbEA2dsOXJsA?w=500&auto=format (qrc:/StatusQ/Components/StatusAnimatedImage.qml:24, )
```

- I think the SFPM is really taking a perf hit. The app is briefly freezing when changing views. E.g. CollectibleDetailView -> CollectiblesView.

- WDYT of including the image loading time in the `loading` state? Currently it doesn't look that good.
<img width="1261" alt="Screenshot 2023-12-05 at 17 19 38" src="https://github.com/status-im/status-desktop/assets/47811206/821e265a-e63b-41b5-acc2-a9b9e6f86452">

caybro commented:
> Nice work! LGTM in general!
> 
> Few findings:
> 
> * The button text needs to be elided
> 
> <img alt="Screenshot 2023-12-05 at 15 42 56" width="807" src="https://private-user-images.githubusercontent.com/47811206/288075828-a2be5b47-8f24-45f9-873c-f0ce1da62d5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE3OTEyMDEsIm5iZiI6MTcwMTc5MDkwMSwicGF0aCI6Ii80NzgxMTIwNi8yODgwNzU4MjgtYTJiZTViNDctOGYyNC00NWY5LTg3M2MtZjBjZTFkYTYyZDVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA1VDE1NDE0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExZTFhYTdjZmJkMDlmMWY5OWIwNTZlMGMxMmRlMzAwZmU4Nzk3NGQwYTc5MzdjNzRkYWM1Nzk4MGMxNDA3ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fYa6JutFgDAUwpxTzBAWlRQa9UCW86ZHGjhdFK93FDc">

Oh yeah, looks like a bug in StatusButton; will try to fix here

> * I've added a few watch accounts and I see some popular NTFs use webp encoding. It seems we can't play webp images from the network.
>   I've been briefly looked into this an I think this is the issue https://bugreports.qt.io/browse/QTBUG-75044. It was fixed in qt6, so it's not helpful in our case.
>   If we really need to support webp formats, I think we need an alternative. It seems to be working fine if we're downloading the image before loading it in the StatusAnimatedImage. @caybro WDYT of this?

Dunno, I don't think we support it 

> ```
> Warning: qrc:/StatusQ/Components/StatusAnimatedImage.qml:24:1: QML StatusAnimatedImage: Error Reading Animated Image File https://i.seadn.io/gae/TX0qOy8g8ibx1p2gQzWIM3Kvsy4a447ti2ZtMlcJmRFM8PU_4YYw2WMpKvO_bnqJiDj2Y7njP6d0tf2YG_K-yYDN2gdbEA2dsOXJsA?w=500&auto=format (qrc:/StatusQ/Components/StatusAnimatedImage.qml:24, )
> ```

Hmm and here it looks like it doesn't like the `query` part of the URL (needs to be stripped maybe)?

> * I think the SFPM is really taking a perf hit. The app is briefly freezing when changing views. E.g. CollectibleDetailView -> CollectiblesView.

Yeah it's not super fast; I'm doing some further changes to it in my followup PR but maybe here the problem is the other component (the detailed view)?

> * WDYT of including the image loading time in the `loading` state? Currently it doesn't look that good.

Hmm, I don't get the question here :)

caybro commented:
@alexjba fixed the warnings, removed the border and fixed the overflowing button:

![image](https://github.com/status-im/status-desktop/assets/5377645/4258127b-fd1e-4389-9904-6377d6162038)

caybro commented on ui/app/AppLayouts/Wallet/views/collectibles/CollectibleView.qml:
```diff
@@ -55,14 +72,14 @@ Control {
             mediaType: root.mediaType
             fallbackImageUrl: root.fallbackImageUrl
             border.color: Theme.palette.baseColor2
             border.width: 1
```
Removed

alexjba commented:
> > * I've added a few watch accounts and I see some popular NTFs use webp encoding. It seems we can't play webp images from the network.
> >   I've been briefly looked into this an I think this is the issue https://bugreports.qt.io/browse/QTBUG-75044. It was fixed in qt6, so it's not helpful in our case.
> >   If we really need to support webp formats, I think we need an alternative. It seems to be working fine if we're downloading the image before loading it in the StatusAnimatedImage. @caybro WDYT of this?
> 
> Dunno, I don't think we support it
> 
> > ```
> > Warning: qrc:/StatusQ/Components/StatusAnimatedImage.qml:24:1: QML StatusAnimatedImage: Error Reading Animated Image File https://i.seadn.io/gae/TX0qOy8g8ibx1p2gQzWIM3Kvsy4a447ti2ZtMlcJmRFM8PU_4YYw2WMpKvO_bnqJiDj2Y7njP6d0tf2YG_K-yYDN2gdbEA2dsOXJsA?w=500&auto=format (qrc:/StatusQ/Components/StatusAnimatedImage.qml:24, )
> > ```
> 
> Hmm and here it looks like it doesn't like the `query` part of the URL (needs to be stripped maybe)?

I've tried it with other urls and I've seen the same result. Not a big deal IMHO because we have a fallback that's loading a static image instead. But it's something we can fix (probably in 0.16 if it's important enough). It's reproducible in storybook as well by adding any webp url as `mediaUrl` and `mediaType: "image"` in the [ManageCollectiblesModel.qml](https://github.com/status-im/status-desktop/pull/12917#diff-8e8e6943f554621d674e54e78277cdd78ea128722719df1730e8f2012781f6b6).

I'll create another issue for it.

alexjba commented:
> > * WDYT of including the image loading time in the `loading` state? Currently it doesn't look that good.
> 
> Hmm, I don't get the question here :)

We have 2 loading states:
- A nice one based on the model `isLoading` [property](https://github.com/status-im/status-desktop/pull/12917/files#diff-8e8e6943f554621d674e54e78277cdd78ea128722719df1730e8f2012781f6b6R54)

<img width="98" alt="Screenshot 2023-12-05 at 18 12 00" src="https://github.com/status-im/status-desktop/assets/47811206/73bb975f-ed80-4b31-a3a5-ba7670d078cd">

- A not so nice one based on the image loading state
<img width="183" alt="Screenshot 2023-12-05 at 18 12 34" src="https://github.com/status-im/status-desktop/assets/47811206/d12bbed6-7cf3-4663-a844-5570befe3a68">

I was thinking to use the first loading state while the image is loading. But it's actually a design decision here.

@benjthayer We can have a few states for the collectible cards:
1. Loading collectible info
2. Loading collectible image
3. Collectible image failed to load

Do we have/need designs for all these states?

alexjba reviewed (approved):
Thanks for the fixes 🍻

noeliaSD commented on ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:
```diff
@@ -1,105 +1,374 @@
+            sourceModel: d.renamedModel
+
+            filters: [
+                ExpressionFilter {
```
Ok!

benjthayer commented:
> @benjthayer We can have a few states for the collectible cards:
> 
> 1. Loading collectible info
> 2. Loading collectible image
> 3. Collectible image failed to load
> 
> Do we have/need designs for all these states?

Thanks @alexjba - we do have the collectible loading state in the Wallet file here which I believe is being used already:
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=13958-473488&mode=design&t=B4czrrKF0XQ9IbCD-4

I'll add the loading state for the community tag to support the new designs unless this is loaded instantaneously?

For loading fails - is it just the image that can fail to load or could all data fail to load?
If we provide a way for the user to refresh / try again, how likely is that action to be successul? Or is the only reason the loading would fail is if say the Opensea API is down and therefore there isn't any action the user can take?

alexjba commented:
> > @benjthayer We can have a few states for the collectible cards:
> > 
> > 1. Loading collectible info
> > 2. Loading collectible image
> > 3. Collectible image failed to load
> > 
> > Do we have/need designs for all these states?
> 
> Thanks @alexjba - we do have the collectible loading state in the Wallet file here which I believe is being used already: https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=13958-473488&mode=design&t=B4czrrKF0XQ9IbCD-4

Indeed, we do and it's used. But this loading state is active while loading the collectible info. Once the collectible info is loaded, the collectible image starts to load from the network and the card looks like this:
<img width="183" alt="Screenshot 2023-12-05 at 18 12 34" src="https://github.com/status-im/status-desktop/assets/47811206/d12bbed6-7cf3-4663-a844-5570befe3a68">

 > I'll add the loading state for the community tag to support the new designs unless this is loaded instantaneously?

IDK much about this. @caybro any idea?

> For loading fails - is it just the image that can fail to load or could all data fail to load?
If we provide a way for the user to refresh / try again, how likely is that action to be successul? Or is the only reason the loading would fail is if say the Opensea API is down and therefore there isn't any action the user can take?

I'm referring only to collectible image failures. If the collectible fails to load (network issue, unsupported format, image provider is down etc. ) the image placeholder is empty. 
Not sure about the reload button. Usually the user can't do much to fix the errors. And the collectible image will be reloaded anyways on several other actions like scrolling or opening other views.

@caybro Do we store in the model collectibles where the info failed to load?
